packages/react-dom/src/client/ReactDOMSelection.js

```diff
@@ -153,11 +157,13 @@ export function getModernOffsetsFromPoints(
  * @param {object} offsets
  */
 export function setOffsets(node, offsets) {
-  if (!window.getSelection) {
+  var doc = node.ownerDocument || document;
```

Comment:
When is ownerDocument ever undefined?

Reply:
According to the MDN page about the `Node` interface:

>  `Node.ownerDocument` returns the `Document` that this node belongs to. If no document is associated with it, returns `null`.

That doesn’t refer to a node that isn’t in the DOM, which will still have a reference to the `ownerDocument` where it originated (via `document.createElement`). In my experience, `ownerDocument` is only `null` for `Document` nodes themselves, which matches what MDN says on the page specifically about `Node.ownerDocument`:

> If this property is used on a node that is itself a document, the result is `null`.

So now that you mention it, this check could be updated to handle `Document` specifically, maybe like `var doc = node.nodeType === 9 ? node : node.ownerDocument;`

Or it could just continue to fallback to the global document whenever `ownerDocument` isn’t truthy.

Reply:
Cool. Thank you. That was extremely helpful!